### PR TITLE
plug security holes and test

### PR DESCRIPTION
**Pull request by CJ-Wright**

### CODE REVIEW

File: conda_forge_tick/env_management.py [hunk 1]
```diff
@@ -0,0 +1,14 @@
+import os
+from contextlib import contextmanager
+
+# SENSITIVE_ENVIRON = {k: os.environ.pop(k, None) for k in ["USERNAME", "PASSWORD", "GITHUB_TOKEN"]}
+SENSITIVE_ENVIRON = {k: os.environ.get(k, None) for k in ["USERNAME", "PASSWORD", "GITHUB_TOKEN"]}
```
Comment (beckermr) [inline conda_forge_tick/env_management.py:5]: We should add `GH_TOKEN` here too.

**Comment by beckermr** [comment]
Why does all of the functionality I think should be there exist in comments now?

**Review by beckermr — CHANGES_REQUESTED** [review]
We'll need to adjust git utils too.

**Review by beckermr — APPROVED** [review]

File: conda_forge_tick/env_management.py [hunk 2]
```diff
@@ -0,0 +1,14 @@
+import os
+from contextlib import contextmanager
+
+SENSITIVE_KEYS = ["USERNAME", "PASSWORD", "GITHUB_TOKEN", "GH-TOKEN"]
```
Comment (isuruf) [inline conda_forge_tick/env_management.py:4]:
```suggestion
SENSITIVE_KEYS = ["USERNAME", "PASSWORD", "GITHUB_TOKEN", "GH_TOKEN"]
```
?

**Review by beckermr — APPROVED** [review]

File: tests/conftest.py [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+
+@pytest.fixture
+def env_setup():
+    old_pwd = os.environ.pop("PASSWORD", None)
+    from conda_forge_tick.env_management import sensitive_env
+
+    with sensitive_env() as env:
+        env["PASSWORD"] = "unpassword"
+        env["PASSWORD2"] = "unpassword"
+    yield
```
Comment (beckermr) [inline tests/conftest.py:14]: Does this yield need to go in one leve?

Comment (beckermr) [reply tests/conftest.py:14]: ahhh nvm

Comment (isuruf) [reply tests/conftest.py:10]: This is not going to work. Removed my commits for now

File: conda_forge_tick/env_management.py [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+
+    @contextmanager
+    def sensitive_env(self):
+        """Add sensitive keys to environ if needed, when ctx is finished remove keys and update the sensitive env
+        in case any were updated inside the ctx"""
+        os.environ.update(**self.clasified_info)
+        yield os.environ
+        self.clasified_info.update(
+            {k: os.environ.pop(k, None) for k in list(self.clasified_info)},
+        )
```
Comment (beckermr) [inline conda_forge_tick/env_management.py:28]: Should use declassify here?

File: conda_forge_tick/env_management.py [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+        self.clasified_info = {}
+        self.classify()
+
+    def classify(self):
+        self.clasified_info.update({k: os.environ.pop(k, None) for k in self.SENSITIVE_KEYS})
+
+    def declassify(self):
+        self.clasified_info.update(
+            {k: os.environ.pop(k, None) for k in list(self.clasified_info)},
+        )
```
Comment (beckermr) [inline conda_forge_tick/env_management.py:18]: Actually looking at this I am confused. doc strings would help me.

Comment (beckermr) [reply conda_forge_tick/env_management.py:18]: declassify is not the opposite of classify. It is doing something else. Maybe `reclassify` or `classify_known_keys`?

**Review by beckermr — APPROVED** [review]

**Comment by CJ-Wright** [comment]
@isuruf ?

**Review by isuruf — APPROVED** [review]

**Pull request merged**